service/src/main/java/uk/gov/hmcts/reform/fpl/service/HearingBookingService.java

```diff
@@ -148,6 +149,11 @@ public HearingBooking getHearingBookingByUUID(List<Element<HearingBooking>> hear
             }).collect(toList());
     }
 
+    public boolean hasFutureHearing(List<Element<HearingBooking>> hearingBookings) {
```

Comment:
Would you say that this method belongs more in CaseData than in this service?
Tbh I think this entire service needs a refactor, methods like `isPastHearing`, `buildHearingBookingElement`, and parts of `buildNewHearingLabel` don't belong in this service and can be relegated to their appropriate models.
Comes down to a discussion myslef and @akomisarek have been having a lot lately about domain logic not being in the domain models.

Reply:
Also by doing this you would remove the dependancy in the content providers of autowiring in what is starting to become quite a heavy service

Reply:
Yup this is a good idea :)

Reply:
but this dependency still in the content providers cause it calls the method hearingService.getMostUrgentHearing

Reply:
😢 
Another method that would be probably doesn't belong where it is, but I can understand not wanting to remove it. Might start becoming more than the original ticket intended

Reply:
Although in case data there is `getMostUrgentHearingBookingAfter`. But would you say this is starting to become more than the ticket originally called for